snapcraft/plugins/jhbuild.py

```diff
@@ -0,0 +1,383 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2016 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""The JHBuild plugin is used for GNOME-based parts.
+
+JHBuild is a tool used to build components of the GNOME ecosystem from version
+control. See the JHBuild manual for more details:
+https://developer.gnome.org/jhbuild/stable/.
+
+The plugin can be customized with the following keys:
+
+  - modules: (required) a list of modules to include in the part, modules that
+             should be skipped should be prefixed with '-', e.g. '-WebKit'
+  - module-set: (default: gnome-world) the module set JHBuild should use
+  - module-set-dir: the directory where the module set is located
+               Either leave unset to use the default moduleset definitions, or
+             set to `.` to reference your project folder.
+  - jhbuild-archive: the source tarball directory on the build host
+  - jhbuild-mirror: the DVCS repository directory on the build host
+
+Advice:
+
+  - You only need to specify a single module in most cases. JHBuild knows the
+    module dependency graph.
+
+  - The desktop-gtk3 helper is incompatible with GTK built by jhbuild. A custom
+    helper is currently required.
+    See github.com/diddledan/corebird for example.
```

Comment:
The problem is that the `desktop-launch` script which gets built by `desktop-gtk3` requires multiple changes to work around the different installation locations of ubuntu's packages vs jhbuild-built packages.